packages/runtime/container-runtime/src/orderedClientElection.ts

```diff
@@ -335,10 +402,16 @@ export class OrderedClientElection
     private addClient(client: ILinkedClient, sequenceNumber: number): void {
         if (this.isEligibleFn(client)) {
             this._eligibleCount++;
-            if (this._electedClient === undefined) {
-                // Automatically elect latest client
+            const isSummarizer = client.client.details.type === summarizerClientType;
+            const electedClientIsSummarizer = this._electedClient?.client.details.type === summarizerClientType;
+            // Note that we allow a summarizer client to supercede an interactive client as elected client.
+            if (this._electedClient === undefined || (isSummarizer && !electedClientIsSummarizer)) {
                 this.tryElectingClient(client, sequenceNumber);
             }
+            else if (this._electedParent === undefined && !isSummarizer) {
```

Comment:
Should this "else" be removed and this condition always be checked? Basically, is it possible that after the "if" condition is executed, _electedParent is undefined? If so, we should try electing the parent too right.

Reply:
No, tryElectingClient will set _electedParent if the new client is interactive, so the "else" is useful.